Branch-SDK/src/io/branch/referral/AnimatedDialog.java

```diff
@@ -0,0 +1,148 @@
+    private void init(Context context) {
+        context_ = context;
+        setDialogWindowAttributes();
+        // Listen for the backpress in order to dismiss the dialog with animation
+        setOnKeyListener(new OnKeyListener() {
+            @Override
+            public boolean onKey(DialogInterface arg0, int keyCode, KeyEvent event) {
+                if (keyCode == KeyEvent.KEYCODE_BACK) {
+                    slideClose();
+                }
+                return true;
+            }
+        });
+    }
+
+    /**
+     * Set the window attributes for the invite dialog.
+     */
+    public void setDialogWindowAttributes() {
+        requestWindowFeature(Window.FEATURE_NO_TITLE);
+        getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_DIM_BEHIND);
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+        WindowManager.LayoutParams lp = new WindowManager.LayoutParams();
+        lp.copyFrom(getWindow().getAttributes());
+        lp.width = WindowManager.LayoutParams.MATCH_PARENT;
+        lp.height = WindowManager.LayoutParams.MATCH_PARENT;
+        lp.gravity = Gravity.BOTTOM;
+        lp.dimAmount = 0.8f;
+        getWindow().setAttributes(lp);
+        getWindow().setWindowAnimations(android.R.anim.slide_in_left);
+        setCanceledOnTouchOutside(true);
+    }
+
+    /**
+     * </p> Opens the dialog with a translation animation to the content view </p>
+     */
+    private void slideOpen() {
+        TranslateAnimation slideUp = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 1.0f, Animation.RELATIVE_TO_SELF, 0f);
+        slideUp.setDuration(500);
+        slideUp.setInterpolator(new AccelerateInterpolator());
+        ((ViewGroup) getWindow().getDecorView()).getChildAt(0).startAnimation(slideUp);
+
+        AnimatedDialog.super.show();
+    }
+
+    /**
+     * </p> Closes the dialog with a translation animation to the content view </p>
+     */
+    private void slideClose() {
+        if (!isClosing_) {
+            TranslateAnimation slideDown = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0.0f, Animation.RELATIVE_TO_SELF, 1f);
+            slideDown.setDuration(500);
+            slideDown.setInterpolator(new DecelerateInterpolator());
+
+            ((ViewGroup) getWindow().getDecorView()).getChildAt(0).startAnimation(slideDown);
+            slideDown.setAnimationListener(new Animation.AnimationListener() {
+                @Override
+                public void onAnimationStart(Animation animation) {
+                    isClosing_ = true;
+                }
+
+                @Override
+                public void onAnimationEnd(Animation animation) {
+                    isClosing_ = false;
+                    AnimatedDialog.super.dismiss();
```

Comment:
It was because we  used override the `Dismiss()` .Since we removed , will be changed  to  `dismiss`